thorlcr/activities/funnel/thfunnelslave.cpp

```diff
@@ -799,16 +798,30 @@ class CNWaySelectActivity : public CSlaveActivity, public CThorSteppable
                     if (whichInput < numRealInputs)
                     {
                         selectedInputITDL = nWayInput->queryConcreteInput(whichInput);
-                        selectedStream = connectSingleStream(*this, selectedInputITDL, 0, selectedJunction, true);  // Should this be passing whichInput??
+                        selectedStream = nWayInput->queryConcreteInputStream(whichInput);
+                        selectedJunction = nWayInput->queryConcreteInputJunction(whichInput);
                         break;
                     }
                     whichInput -= numRealInputs;
                 }
+                else
+                {
+                    if (whichInput == 0)
+                    {
+                        selectedInputITDL = cur;
+                        selectedStream = queryInputStream(i);
+                        selectedJunction = queryInputJunction(i);
+                    }
+                    whichInput -= 1;
```

Comment:
Not new, but yes.
It will break out on line below, but should have a break in block above.
I'll change.